diff -rup cereal-1.2.2/include/cereal/archives/json.hpp.old cereal-1.2.2/include/cereal/archives/json.hpp
--- cereal-1.2.2/include/cereal/archives/json.hpp.old	2017-10-15 22:10:56.000000000 -0700
+++ cereal-1.2.2/include/cereal/archives/json.hpp	2017-10-15 22:11:12.000000000 -0700
@@ -477,7 +477,7 @@ namespace cereal
           }
 
           Iterator(ValueIterator begin, ValueIterator end) :
-            itsValueItBegin(begin), itsValueItEnd(end), itsIndex(0), itsType(Value)
+            itsValueItBegin(begin), itsIndex(0), itsType(Value)
           {
             if( std::distance( begin, end ) == 0 )
               itsType = Null_;
@@ -532,7 +532,7 @@ namespace cereal
 
         private:
           MemberIterator itsMemberItBegin, itsMemberItEnd; //!< The member iterator (object)
-          ValueIterator itsValueItBegin, itsValueItEnd;    //!< The value iterator (array)
+          ValueIterator itsValueItBegin;                   //!< The value iterator (array)
           size_t itsIndex;                                 //!< The current index of this iterator
           enum Type {Value, Member, Null_} itsType;        //!< Whether this holds values (array) or members (objects) or nothing
       };
