diff --git configure configure
index 94f0f27..982fe53 100755
--- configure
+++ configure
@@ -16955,7 +16955,7 @@ fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_lib_lzma_lzma_mf_is_supported" >&5
 $as_echo "$ac_cv_lib_lzma_lzma_mf_is_supported" >&6; }
 if test "x$ac_cv_lib_lzma_lzma_mf_is_supported" = xyes; then :
-  LIBLZMA=-lzma
+  LIBLZMA=-llzma
 
 
 $as_echo "#define HAVE_LZMA 1" >>confdefs.h
